id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoServiceImpl.java

```diff
@@ -533,11 +535,15 @@ private void updateRequestBodyData(IdRequestDTO request) throws IdRepoAppExcepti
 	@SuppressWarnings({ "rawtypes", "unchecked" })
 	protected void updateVerifiedAttributes(IdRequestDTO<T> requestDTO, DocumentContext inputData, DocumentContext dbData) throws IdRepoAppException {
 		List dbVerifiedAttributes = (List) dbData.read(DOT + VERIFIED_ATTRIBUTES);
+		boolean isV2Version=false;
 		dbVerifiedAttributes.remove(null);
-		boolean isV2Version = requestDTO.getVerifiedAttributes() instanceof Map ? true : false;
+		if (requestDTO.getVerifiedAttributes() != null) {
+			isV2Version = requestDTO.getVerifiedAttributes() instanceof Map;
```

Comment:
Can we not pass the v2 flag as the method argument based on the method invoked in the controller?